cmd/in/main.go

```diff
@@ -120,25 +165,38 @@ func get(req InRequest, ref name.Reference, dest string) error {
 		imageOpts = append(imageOpts, remote.WithAuth(auth))
 	}
 
-	image, err := remote.Image(ref, imageOpts...)
+	image, err := remote.Image(digest, imageOpts...)
 	if err != nil {
-		return fmt.Errorf("locate remote image: %w", err)
+		return nil, fmt.Errorf("locate remote image: %w", err)
+	}
+	if image == empty.Image {
```

Comment:
@owenfarrell Do you remember what this is about? Noticed it while refactoring and can't see any tests that cover it. Do mirror registries sometimes return empty images?

Reply:
@vito - I think that conditional can go.

I originally added it in support of the retry logic in place to support a 429 response. But when testing against a local (dummy) "mirror", I realized that the manifest fetch and blob fetch are really 2 different transactions with a checkpoint in between.

So to make the [test sequence](https://github.com/concourse/registry-image-resource/pull/183/files/8e96807508d46b29c8265c033a07cf5d6533c783#diff-bad64590066f2bc6f879c51275e6a399R398) realistic, I took those requests out... but I never removed that conditional.